src/Signature/SignatureProvider.php

```diff
@@ -114,6 +114,8 @@ public static function version()
                     return $service === 's3'
                         ? new S3SignatureV4($service, $region)
                         : new SignatureV4($service, $region);
+                case 'v4-unsigned-body':
+                    return new SignatureV4($service, $region, $unsigned = true);
```

Comment:
I think the `$unsigned = ` here is unnecessary. If you feel an additional positional parameter makes the constructor less clear, it would be more in line with the rest of the SDK to add an `$options` array (containing an `'unsigned_payload' => true` member here) as the final parameter. That would allow more options to be added over time.